students/PhongT/lesson04/mailroom_part2.py

```diff
@@ -0,0 +1,142 @@
+least five donors,  with between 1 and 3 donations each.
+"""
+donors_data = [
+        {"name": "Ken Lenning", "donations": [1325, 1232.24, 16325.5]},
+        {"name": "Thomas Timmy","donations": [521.3, 869.14]},
+        {"name": "Jane Chung", "donations": [3259.3, 1282.74, 1525.5]},
+        {"name": "Lucy Nguyen", "donations": [5253.82]},
+        {"name": "Ben Cormack", "donations": [56230.25, 89532.0, 12025.8]}]
+
+
+def print_thank_you():
+    """ Add or Update a donor record, then print a Thank You letter """
+    while True:
+        print()
+        response = input(''' sub_menu --- Donation Entries ---
+        Type your Full Name,
+        or 'list' to display list of the donor names,
+        or 'quit' to quit and return to main program prompt.
+        Enter your choice here: > ''')
+
+        if response.lower().strip() in ["q", "quit"]:
+            break
+        elif response.lower().strip() in ["l", "list"]:
+            print_donor_list()
+            continue
+        else:
+            record_id = None
+            for index, existing_donor in enumerate(donors_data):
+                # get a match an existing record
+                if existing_donor["name"].lower().strip() == response.lower().strip():
+                    record_id = index
+                    break
+
+            amount = input("Enter Donation amount: ")
+            try:
+                donate_amount = float(amount)
+            except ValueError:
+                print("The donation amount must be numeric!")
+
+            # add or update a record
+            if record_id is None:  # new record
+                print('Add a new record entry')
+                donors_data.append({"name": response, "donations": [donate_amount]})
+                donor_to_print = donors_data[-1]
+            else:  # existing record
+                print('Update an existing record entry')
+                donors_data[record_id]["donations"].append(donate_amount)
+                donor_to_print = donors_data[record_id]
+
+            # now print a thank you letter
+            print_letter(donor_to_print)
+
+
+def print_letter(donor):
+    """ Generate a thank you note to a donor """
+    msg = get_thankyou_message(donor)
+    print(msg)
+
+
+def get_thankyou_message(donor):
+    """ generate a formatted thank you letter message"""
+    message = '''Dear {}, 
+            Thank you so much for your generosity with your most recent donation of ${}. 
+            It will be put to very good use.
+            Sincerely.'''
+    return message.format(donor["name"], donor["donations"][-1])
+
+
+def print_donor_list():
+    """ print out list of current donors"""
+    print('Below are the existing donors: ')
+    for donor in donors_data:
+        print('\t- ', donor["name"], ' ', donor["donations"])
+
+
+def print_report():
+    """ print a list of your donors, sorted by total historical donation amount. """
+    width = 68
+    print("-" * width)
```

Comment:
since `width` isnt changing, perhaps:
`dash_line = ("-" * 68)`
and just use that going forward as:
`print(dash_line)`